keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ClusterDeployStrategy.kt

```diff
@@ -55,7 +57,8 @@ data class RedBlack(
       (waitForInstancesUp ?: DEFAULT_WAIT_FOR_INSTANCES_UP) +
         (delayBeforeDisable ?: ZERO) +
         (delayBeforeScaleDown ?: ZERO)
-      ).toMillis()
+      ).toMillis(),
+    "interestingHealthProviderNames" to if (considerOnlyAmazonHealth) listOf("Amazon") else null
```

Comment:
This should also be generic, I think the specific cloud provider should be passed in by the specific handler

Reply:
Honestly this whole thing should be an extension method inside of the handler rather than implemented here